src/commands/deploy.js

```diff
@@ -220,6 +222,9 @@ const runDeploy = async ({
       filter: getDeployFilesFilter({ site, deployFolder }),
     })
   } catch (e) {
+    if (deployId) {
```

Comment:
[sand] Should this be an additional `try`/`catch` block around `deployEdgeHandlers()` + `api.deploy()`? Like this, there would be no need to check if `deployId` exists.

Reply:
That would mean we need to re-throw the error after canceling the deploy to reach the other part of the error handling code (we could also refactor that part when we do https://github.com/netlify/js-client/issues/157 to have a dedicated `try/catch` per operation to have better error reporting).

Reply:
Going to merge and release. We can discuss this in https://github.com/netlify/js-client/issues/157

Reply:
Makes sense!